From: Dimitri John Ledkov <xnox@ubuntu.com>
Date: Mon, 19 Feb 2018 20:47:41 +0000
Subject: test/test-functions: on PP64 use vmlinux

At least on Ubuntu, ppc64el uses vmlinux-, not vmlinuz. With this, it should be
possible to run qemu tests on ppc64el as part of Ubuntu autopkgtests.

(cherry picked from commit a2ab2bdd5fcbd15c1f9daf4eb34c4dfb56c12e30)
---
 test/test-functions | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/test/test-functions b/test/test-functions
index ac3de81..87235e3 100644
--- a/test/test-functions
+++ b/test/test-functions
@@ -72,7 +72,15 @@ run_qemu() {
         if [[ "$LOOKS_LIKE_ARCH" ]]; then
             KERNEL_BIN=/boot/vmlinuz-linux
         else
-            KERNEL_BIN=/boot/vmlinuz-$KERNEL_VER
+            [ "$ARCH" ] || ARCH=$(uname -m)
+            case $ARCH in
+                ppc64*)
+                KERNEL_BIN=/boot/vmlinux-$KERNEL_VER
+                ;;
+                *)
+                KERNEL_BIN=/boot/vmlinuz-$KERNEL_VER
+                ;;
+            esac
         fi
     fi
 
